datatypes/src/primitives/feature_data.rs

```diff
@@ -443,6 +469,216 @@ fn null_bitmap_to_bools(null_bitmap: &Option<Bitmap>, len: usize) -> Vec<bool> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct BoolDataRef<'f> {
+    buffer: Vec<bool>,
```

Comment:
could this also be a bitmap?

Reply:
At the moment this cannot be a bitmap, because DataRef requires the implementation of AsRef<[bool]>. Arrow internally uses bit packing to store a bitmap. As a result of this the packed data must be first converted so a bool slice can be created. This is not possible inside of AsRef, because the generated Vec would get dropped.